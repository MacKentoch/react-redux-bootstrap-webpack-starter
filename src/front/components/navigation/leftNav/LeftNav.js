// @flow

// #region imports
import React from 'react';
import LeftNavButton from './leftNavButton/LeftNavButton';
// #endregion

// #region flow types
export type LeftLink = {
  link: string,
  label: string,
  viewName: string,
};

export type OnLeftNavButtonClick = (
  event: SyntheticEvent<>,
  viewName: string,
) => any;

type Props = {
  leftLinks: Array<LeftLink>,
  onLeftNavButtonClick: OnLeftNavButtonClick,

  ...any,
};
// #endregion

const LeftNav = ({ leftLinks, onLeftNavButtonClick }: Props) => {
  return (
    <ul className="nav navbar-nav">
      {leftLinks.map(({ link, label, viewName }, index) => {
        return (
          <LeftNavButton
            key={index}
            link={link}
            label={label}
            viewName={viewName}
            onClick={onLeftNavButtonClick}
          />
        );
      })}
    </ul>
  );
};

// #region statics
LeftNav.defaultProps = { leftLinks: [] };
LeftNav.displayName = 'LeftNav';
// #endregion

export default LeftNav;
