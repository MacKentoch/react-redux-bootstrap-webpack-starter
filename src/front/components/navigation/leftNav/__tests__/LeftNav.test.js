// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import LeftNav from '../LeftNav';
import { type LeftLink, type OnLeftNavButtonClick } from '../LeftNav';
// #endregion

// #region flow types
type Props = {
  leftLinks?: Array<LeftLink>,
  onLeftNavButtonClick?: OnLeftNavButtonClick,
};
// #endregion

describe('LeftNav component', () => {
  it('renders as expected', () => {
    const props: Props = {
      leftLinks: [
        { link: '1', label: 'view 1', viewName: 'view 1' },
        { link: '2', label: 'view 2', viewName: 'view 2' },
      ],
      onLeftNavButtonClick: () => {},
    };

    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <LeftNav {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
