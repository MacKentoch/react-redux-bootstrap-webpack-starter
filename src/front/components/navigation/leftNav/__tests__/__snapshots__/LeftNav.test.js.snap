// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav component renders as expected 1`] = `
<div>
  <ul
    className="nav navbar-nav"
  >
    <li>
      <a
        href="1"
        onClick={[Function]}
      >
        view 1
      </a>
    </li>
    <li>
      <a
        href="2"
        onClick={[Function]}
      >
        view 2
      </a>
    </li>
  </ul>
</div>
`;
